### PR TITLE
Apply patches from OSGeo4W upstreams backported to relbr78

### PR DESCRIPTION
**Pull request by landam**
This PR introduces #1880 into release branch 7.8.

### CODE REVIEW

File: mswindows/GRASS-Installer.nsi.tmpl [hunk 1]
```diff
@@ -896,22 +839,8 @@ Section "Important Microsoft Runtime DLLs" SecMSRuntime
...
-            StrCpy $ARCHIVE_NAME2015 "msvcrt2015-1.0-1.tar.bz2"
-        !else
-            StrCpy $ARCHIVE_NAME "msvcrt-1.0.1-14.tar.bz2"
-            ;StrCpy $ARCHIVE_NAME2005 "msvcrt2005-1.0-1.tar.bz2"
-            StrCpy $ARCHIVE_NAME2008 "msvcrt2008-1.0-1.tar.bz2"
-            StrCpy $ARCHIVE_NAME2010 "msvcrt2010-1.0-1.tar.bz2"
-            StrCpy $ARCHIVE_NAME2013 "msvcrt2013-1.0-1.tar.bz2"
-            StrCpy $ARCHIVE_NAME2015 "msvcrt2015-1.0-1.tar.bz2"
-        !endif
+	StrCpy $HTTP_PATH "http://download.osgeo.org/osgeo4w/v2/${PLATFORM}/release/msvcrt2019/"
```
Comment (hellik) [inline mswindows/GRASS-Installer.nsi.tmpl:842]: where do we define **${PLATFORM}**?

**Review by hellik — COMMENTED** [review]
I've left one comment there.

File: mswindows/GRASS-Installer.nsi.tmpl (hunk 1 above)
Comment (landam) [reply mswindows/GRASS-Installer.nsi.tmpl:842]: @hellik https://github.com/OSGeo/grass/blob/releasebranch_7_8/mswindows/GRASS-Installer.nsi.tmpl#L45

I tried to keep modifications in releasebranch_7_8 minimalistic. I will prepare separated PR for main branch.

**Comment by landam** [comment]
You can test first standalone installer build on the top of OSGeo4W v2: https://wingrass.fsv.cvut.cz/grass78/x86_64/WinGRASS-7.8.6dev-7587f94e0-656-Setup-x86_64.exe

**Comment by landam** [comment]
There is one issue: the size of standalone installer increased almost by 500MB:

```
WinGRASS-7.8.6dev-7587f94e0-656-Setup-x86_64.exe        2021-09-29 18:24  659M 
WinGRASS-7.8.6dev-655-Setup-x86_64.exe                  2021-09-27 22:37  184M 
```

This is caused by a new dependency: `proj-data` (see http://download.osgeo.org/osgeo4w/v2/x86_64/release/grass/setup.hint)

**Comment by landam** [comment]
You can test first standalone installer build on the top of osgeo4w v2: https://wingrass.fsv.cvut.cz/grass78/x86_64/WinGRASS-7.8.6dev-7587f94e0-656-Setup-x86_64.exe

**Comment by landam** [comment]
There is a new issue: the size of installer increased by 500MB(!):

```
WinGRASS-7.8.6dev-7587f94e0-656-Setup-x86_64.exe        2021-09-29 18:24  659M  
WinGRASS-7.8.6dev-655-Setup-x86_64.exe                  2021-09-27 22:37  184M  
```

This is caused by new dependecy (```proj81-runtime -> proj-data```). But this cannot be changed easily (`proj-data` is dependency for `proj81-runtime`).

**Comment by landam** [comment]
If no objections I will merge this PR in the next days.

**Review by ninsbl — APPROVED** [review]
Looks good to me. I have no experience with the packaging part though.

Do you know if the gisinit part of the patch from OSGeo4W that you did not apply may affect how AddOns work? Are AddOns only build agains the released version  / point release?

**Comment by ninsbl** [comment]
A nice feature of OSGeo4W Version 2 is that it does not require administrator rights. Would it be possible to make the installer run without elevated priveleges?

**Comment by landam** [comment]
> Do you know if the gisinit part of the patch from OSGeo4W that you did not apply may affect how AddOns work? Are AddOns only build agains the released version / point release?

The change in gisinit part not related to Addons but to QGIS GRASS plugin. Addons are built always against specific point version. Moreover gisinit check doesn't relay on point version. It means that QGIS doesn't need to be re-build when a new point version of GRASS is released. I think it could be removed from osgeo4w patches.

**Comment by hellik** [comment]
> 
> 
> A nice feature of OSGeo4W Version 2 is that it does not require administrator rights. Would it be possible to make the installer run without elevated priveleges?

AFAIK you can't install software in _C:\Program Files_ without elevated privileges.

**Comment by hellik** [comment]
> 
> 
> You can test first standalone installer build on the top of osgeo4w v2: https://wingrass.fsv.cvut.cz/grass78/x86_64/WinGRASS-7.8.6dev-7587f94e0-656-Setup-x86_64.exe

installing standalone works, though a little bit slow due to the large file size.

do we have any idea if we could sign the installer, that MS windows is happy without telling you that the software may be harmful? 

QGIS does this e.g. for their apple binaries.

**Comment by hellik** [comment]
> 
> 
> If no objections I will merge this PR in the next days.

no objections, go ahead

**Pull request merged**